apps/src/p5lab/spritelab/blocks.js

```diff
@@ -352,6 +377,64 @@ const customInputTypes = {
       )}'}`;
     }
   },
+  behaviorPicker: {
+    addInput(blockly, block, inputConfig, currentInputRow) {
+      currentInputRow
+        .appendField(inputConfig.label)
+        .appendField(
+          new Blockly.FieldDropdown(getAllBehaviors, undefined, undefined),
+          inputConfig.name
+        );
+      let allowBehaviorEditing = Blockly.useModalFunctionEditor;
+      if (
+        window.appOptions && // global appOptions is not available on level edit page
+        appOptions.level.toolbox &&
+        !appOptions.readonlyWorkspace &&
+        !Blockly.hasCategories
+      ) {
+        allowBehaviorEditing = false;
+      }
+      if (allowBehaviorEditing) {
+        const editLabel = new Blockly.FieldIcon(Blockly.Msg.FUNCTION_EDIT);
+        Blockly.cdoUtils.bindBrowserEvent(
+          editLabel.fieldGroup_,
+          'mousedown',
+          block,
+          this.openEditor
+        );
+        currentInputRow.appendField(editLabel);
+      }
+    },
+    generateCode(block, arg) {
+      const invalidBehavior =
+        block.getFieldValue(arg.name) ===
+        Blockly.FieldDropdown.NO_OPTIONS_MESSAGE;
+      const behaviorId = Blockly.JavaScript.variableDB_?.getName(
+        block.getFieldValue(arg.name),
+        'PROCEDURE'
+      );
+      if (invalidBehavior) {
+        console.warn('No behaviors available');
```

Comment:
should we be logging to the console here?

Reply:
Yes, that's where it ends up. FWIW, the CSC team currently make extensive use of the dev console for things like validation and helper code: https://levelbuilder-studio.code.org/s/csc-mappinglandmarks-2023/lessons/3/levels/1
This was mostly something I added for their sake.
![image](https://user-images.githubusercontent.com/43474485/225123435-2cf766b6-9f7d-47b5-b8fc-f348604a5809.png)

Reply:
ok if it's useful for levelbuilders we can leave it in then :)